src/Compilers/Core/Portable/MetadataReader/PEModule.cs

```diff
@@ -3613,23 +3623,17 @@ private ConstantValue GetConstantValueOrThrow(ConstantHandle handle)
 
             if (ignoreCase)
             {
-                // This linear search is not the optimal way to use a hashmap, but we should only use
-                // this functionality when computing diagnostics.  Note
-                // that we can't store the map case-insensitively, since real metadata name
-                // lookup has to remain case sensitive.
-                foreach (var pair in _lazyForwardedTypesToAssemblyIndexMap)
+                ensureCaseInsensitiveDictionary();
+
+                if (_lazyCaseInsensitiveForwardedTypesToAssemblyIndexMap.TryGetValue(fullName, out var value))
                 {
-                    if (string.Equals(pair.Key, fullName, StringComparison.OrdinalIgnoreCase))
-                    {
-                        matchedName = pair.Key;
-                        return pair.Value;
-                    }
+                    matchedName = fullName;
```

Comment:
>matchedName = fullName [](http://example.com/codeflow?start=20&length=22)

It looks like there is a behavior change here, we are no longer returning the matched casing.